src/types.jl

```diff
@@ -566,6 +566,116 @@ function System(;
                     loggers, k_converted, force_units, energy_units, atom_masses)
 end
 
+"""
+    System(crystal; <keyword arguments>)
+
+A physical system to be simulated, constructed from a SimpleCrystals.jl `Crystal` struct.
+Properties unused in the simulation or in analysis can be left with their
+default values.
+`atoms`, `atoms_data`, `coords` and `boundary` are automatically calcualted from the cyrstal struct. Extra
+atom paramaters like `σ` will have to be added manually after construction using the convenience constructor:
+System(system; <keyword arguments>)
+
+
+This is a sub-type of `AbstractSystem` from AtomsBase.jl and implements the
+interface described there.
+
+    
+
+# Arguments
+"""
+function System(crystal::Crystal{D};
+        pairwise_inters=(),
+        specific_inter_lists=(),
+        general_inters=(),
+        constraints=(),
+        velocities=nothing,
+        neighbor_finder=NoNeighborFinder(),
+        loggers=(),
+        force_units=u"kJ * mol^-1 * nm^-1",
+        energy_units=u"kJ * mol^-1",
+        k=Unitful.k) where D
+
+    # Parse atom data that isbits()
+    atoms = [Atom(index=i, charge=a.charge, mass=a.mass) for (i,a) in enumerate(crystal.atoms)]
+
+    # Parse other atom data
+    atoms_data = [AtomData(element = String(a.sym),) for a in crystal.atoms]
+
+    coords = SimpleCrystals.position(crystal, :)
+
+    # Build bounding box
+    side_lengths = norm.(eachrow(bounding_box(crystal)))
+    if any(typeof(crystal.lattice.crystal_family) .<: [CubicLattice, OrthorhombicLattice, TetragonalLattice])
+        boundary = CubicBoundary(side_lengths...)
+    elseif any(typeof(crystal.lattice.crystal_family) .<: [SquareLattice, RectangularLattice])
+        boundary = RectangularBoundary(side_lengths...)
+    elseif D == 2 #Honeycomb, Hex2D, & Oblique
+        throw(error("$(crystal.lattice.crystal_family) is not supported as it would need a 2D triclinic boundary.
+             Try defining the crystal with a rectangular or square unit cell."))
+    else #3D non-cubic systems
+        @assert all(crystal.lattice.crystal_family.lattice_angles .< 90u"°") "All crystal lattice angles must be less than 90°"
```

Comment:
`ArgumentError` if this is a valid crystal that the constructor does not support, `error` otherwise.